jenkins/spark-tests.sh

```diff
@@ -215,7 +215,9 @@ run_iceberg_tests() {
   # get the major/minor version of Spark
   ICEBERG_SPARK_VER=$(echo $SPARK_VER | cut -d. -f1,2)
   IS_SPARK_33_OR_LATER=0
-  [[ "$(printf '%s\n' "3.3" "$ICEBERG_SPARK_VER" | sort -V | head -n1)" = "3.3" ]] && IS_SPARK_33_OR_LATER=1
```

Comment:
Curious why the existing code doesn't already handle this properly?  Seems to do the right thing as-is:
```bash
$ printf '%s\n' "3.3" "3.4" | sort -V | head -n1
3.3
```